mathics/builtin/integer.py

```diff
@@ -136,3 +136,338 @@ def apply(self, n, b, evaluation):
 
         result = sympy.Integer(sympy.log(abs(n), b)) + 1
         return Integer(result)
+
+
+class BitLength(Builtin):
+    """
+    <dl>
+    <dt>'BitLength[$x$]'
+        <dd>gives the number of bits needed to represent the integer $x$. $x$'s sign is ignored.
+    </dl>
+
+    >> BitLength[1023]
+     = 10
+    >> BitLength[100]
+     = 7
+    >> BitLength[-5]
+     = 3
+    >> BitLength[0]
+     = 0
+    """
+
+    def apply(self, n, evaluation):
+        'BitLength[n_Integer]'
+        return Integer(n.get_int_value().bit_length())
```

Comment:
This implements what I would say is the expected behaviour. However, mathematica behaves strangely for negative ints, see https://reference.wolfram.com/language/ref/BitLength.html.

In short, I think a correct implementation is
```
n = n.get_int_value()
if n < 0:
    n = -1 - n
return Integer(n.bit_length())
```

so that e.g. `BitLength[-1]` is `0`.